vita3k/kernel/src/thread/thread.cpp

```diff
@@ -117,16 +117,18 @@ SceUID create_thread(Ptr<const void> entry_point, KernelState &kernel, MemState
 
     alloc_name = fmt::format("TLS for thread {} (#{})", name, thid);
 
-    constexpr auto tls_size = 0x800;
-    auto tls_address_ptr = Ptr<void>(alloc(mem, tls_size, alloc_name.c_str()));
-    auto tls_address = tls_address_ptr.address() + tls_size;
+    auto tls_size = kernel_tls_size + kernel.tls_msize;
+    auto base_tls_address_ptr = Ptr<uint8_t>(alloc(mem, tls_size, alloc_name.c_str()));
+    memset(base_tls_address_ptr.get(mem), 0, tls_size);
+    thread->tls = std::make_shared<Resource<Address>>(base_tls_address_ptr.address(), stack_deleter);
```

Comment:
Don't think making this shared is good idea. Just using normal Resource should work just fine. Even if it doesn't work we better make this unique_ptr, because the owner is apparent.

Reply:
If Resource is not used by many places, we can just get rid of its copy constructor. That way we can avoid dealing with hassles of return value optimizations etc.

Reply:
I'll dig into this Resource thing and pr maybe tommorow. I have some ideas.

Reply:
Yes, but I don't know how to write it working. changing to 
struct ThreadState {
    ....
    Resource<Address> tls;
};
////
    thread->tls = Resource<Address>(base_tls_address_ptr.address(), stack_deleter);
does not compile

Reply:
https://github.com/Vita3K/Vita3K/pull/1258

Here we go